### ISSUE TITLE
CanvasGroup with ShaderMaterial renders white rectangle instead of expected texture

### ISSUE THREAD

**Issue by firebelley** [godotengine/godot#69885]
### Godot version

v4.0.beta8.mono.official [45cac42c0]

### System information

Windows 11, NVIDIA RTX 3070, Vulkan Forward+

### Issue description

When adding a ShaderMaterial to a `CanvasGroup` it renders a white rectangle instead of the expected texture (in this case, the Godot icon).

![image](https://user-images.githubusercontent.com/4622925/206880269-b027e73c-aad6-499e-8b76-679d1eef753f.png)

The shader should just be using the child sprite 2D's texture:

```
shader_type canvas_item;

void fragment() {
	COLOR = texture(TEXTURE, UV);
}
```

Note that attaching a shader is not necessary. The error happens as soon as a `ShaderMaterial` is added to the `CanvasGroup`.

This is on Vulkan with the `forward_plus` rendering method. The error does not occur after switching to the `gl_compatibility` rendering method in the project settings.

Also worth noting is this issue was present in Beta 7 as well.

### Steps to reproduce

- Add a `CanvasGroup`
- Add a `ShaderMaterial` to the `CanvasGroup` node and observe issue

### Minimal reproduction project

[Canvas Group MRP.zip](https://github.com/godotengine/godot/files/10201228/Canvas.Group.MRP.zip)

**Comment by clayjohn** [comment]
This is expected behaviour. By adding your own shader you are overriding the CanvasGroup shader. This is the same thing that happens when you add a custom shader to a label (or any other node with text). You become responsible for the shader when you add your own shader. 

In this case, you should be using ``SCREEN_TEXTURE`` and ``SCREEN_UV`` instead of ``TEXTURE`` and ``UV`` if you want to read from the child. ``TEXTURE`` contains the node's texture which defaults to white since there is no texture defined. 

We should add a warning here: https://docs.godotengine.org/en/latest/classes/class_canvasgroup.html

and perhaps even here https://docs.godotengine.org/en/latest/tutorials/shaders/shader_reference/canvas_item_shader.html

**Comment by firebelley** [comment]
Gotcha, thanks. Just tried it in the MRP and it works.

Why does it work like I originally showed with gl_compatibility mode then? Is that a bug?

**Comment by clayjohn** [comment]
> Gotcha, thanks. Just tried it in the MRP and it works.
> 
> Why does it work like I originally showed with gl_compatibility mode then? Is that a bug?

Yes, it looks like the gl_compatibility renderer is ignoring your shader altogether. For example if you change it to:
```
shader_type canvas_item;

void fragment() {
	COLOR = vec4(1.0, 0.0, 0.0, 1.0);
}
```

Nothing happens

**Comment by Crystalwarrior** [comment]
This is still not fixed in Godot 4.1 dev3.
